app/models/chess_piece.rb

```diff
@@ -10,6 +10,16 @@ def vertical_move?(endpoint_x, endpoint_y)
     (position_y != endpoint_y) && (position_x === endpoint_x)
   end
 
+  def vertically_obstructed?(endpoint_x, endpoint_y)
+    is_obstructed = false
+    ((position_y + 1)..endpoint_y).find do |y|
+      if game.piece_in_square(endpoint_x, y)
```

Comment:
you have `piece_in_square` where you're defining the method `piece_in_square?`, you're missing the `?`